pkg/kn/flags/podspec_helper.go

```diff
@@ -398,6 +399,152 @@ func UpdateSecurityContext(spec *corev1.PodSpec, securityContext string) error {
 	return nil
 }
 
+// UpdateNodeSelector updates the Node Selector
+func UpdateNodeSelector(spec *corev1.PodSpec, nodeSelector []string) error {
+	nodeSelectorsAllMap, err := util.MapFromArrayAllowingSingles(nodeSelector, "=")
+	if err != nil {
+		return err
+	}
+	nodeSelectorsToRemove := util.ParseMinusSuffix(nodeSelectorsAllMap)
+	nodeSelectorNew := spec.NodeSelector
+	if nodeSelectorNew == nil {
+		nodeSelectorNew = map[string]string{}
+	}
+	for key, value := range nodeSelectorsAllMap {
+		nodeSelectorNew[key] = value
+	}
+	for _, key := range nodeSelectorsToRemove {
+		delete(nodeSelectorNew, key)
+	}
+	spec.NodeSelector = nodeSelectorNew
+	return nil
+}
+
+// UpdateTolerations updates the configuration for Tolerations
+func UpdateTolerations(spec *corev1.PodSpec, toleration []string) error {
+	tolerationsExisting := spec.Tolerations
+	if tolerationsExisting == nil {
+		tolerationsExisting = []corev1.Toleration{}
+	}
+	tolerationNew := corev1.Toleration{}
+	tolerationsAllMap, err := util.MapFromArray(toleration, "=")
+	if err != nil {
+		return err
+	}
+	for key, value := range tolerationsAllMap {
+		switch strings.ToLower(key) {
+		case "key":
+			tolerationNew.Key = value
+		case "value":
+			tolerationNew.Value = value
+		case "effect":
+			if value == string(v1.TaintEffectNoSchedule) {
+				tolerationNew.Effect = v1.TaintEffectNoSchedule
+			} else if value == string(v1.TaintEffectPreferNoSchedule) {
+				tolerationNew.Effect = v1.TaintEffectPreferNoSchedule
+			} else if value == string(v1.TaintEffectNoExecute) {
+				tolerationNew.Effect = v1.TaintEffectNoExecute
+			}
+		case "operator":
+			if value == string(v1.TolerationOpExists) {
+				tolerationNew.Operator = v1.TolerationOpExists
+			} else if value == string(v1.TolerationOpEqual) {
+				tolerationNew.Operator = v1.TolerationOpEqual
+			}
+		}
+	}
+	tolerationsExisting = append(tolerationsExisting, tolerationNew)
+	spec.Tolerations = tolerationsExisting
+
+	return err
+}
+
+// UpdateNodeAffinity updates the configuration for Node Affinity.
+func UpdateNodeAffinity(spec *corev1.PodSpec, nodeAffinity []string) error {
+	var matchExpressionsExisting []v1.NodeSelectorRequirement
+	var nodeSelectorTermsExisting []v1.NodeSelectorTerm
+	var preferenceExisting []v1.PreferredSchedulingTerm
+	var preferenceWeight int32
+	matchExpressionNew := v1.NodeSelectorRequirement{}
+	nodeAffinityAllMap, err := util.MapFromArray(nodeAffinity, "=")
+	if err != nil {
+		return err
+	}
+	var nodeAffinityType string
+	for key, value := range nodeAffinityAllMap {
+		if strings.ToLower(key) == "type" {
+			if value == "Required" {
+				nodeAffinityType = "Required"
```

Comment:
It's a small nit. I would probably make it less restrictive and with `strings.ToLower(value) == "required"`. We have that at other places that parse such inputs. But it doesn't need to addressed immediately.